src/io/jesi/customs/strict.cljc

```diff
@@ -40,41 +40,42 @@
   matches (with `re-find`) the regular expression `re`."
   ([form]
    {:pre [(some? form)]}
-   `(env/transform
-      (test/is ~form)))
+   (let [is (env/symbol &env `test/is)]
```

Comment:
`StackOverflowError` still occurs even when not using `env/transform`, but using `env/symbol` should lead to faster compile times at least. The workaround is to create a cljs specific version of strict (not sure why is works). See `jesi-cljs/test/io/jesi/cljs/test/strict.cljc`